coffea/processor/executor.py

```diff
@@ -216,6 +216,42 @@ def __call__(self, items):
         return accumulate(items)
 
 
+class FuturesHolder:
+    def __init__(self, futures, refresh=2):
+        self.futures = set(futures)
+        self.merges = set()
+        self.completed = set()
```

Comment:
Won't this hold a reference to all task futures until the processing has finished? In which case all the intermediate output stays alive longer than it needs to. This would be really bad for local processing.

Reply:
Well technically with merging the intermediate/finished futures should get cleaned up. OTOH the merge futures get queue only after the standard jobs, not sure if there's a way around that. 

The future itself should add negligible overhead, not sure how bad we are doing with multiple hists vs. merged hist.

Reply:
Ok, added intermediate merging on the submit node when merge jobs are turned off. Should be functionally equivalent to current implementation

Reply:
Futures hold internally a ref to the result, which is large (histograms)
But if the goal here is to only hold `completed` that are pending a merge job and `merges` only holds finished merges pending a final merge, then I think that's ok. We might be in danger of having too deep a queue here though, but maybe lets wait and see if it becomes a problem